create_plots.py

```diff
@@ -36,5 +36,230 @@ def plot_gpsc_counts(df: pd.DataFrame):
+        order=df["In_silico_ST"].value_counts().index,
+        palette=colours,
+        stat="percent",
+        hue="In_silico_ST",
+    )
+
+    # add the x and y axis labels
+    plt.xlabel("In_silico_ST")
+    plt.ylabel("Percentage")
+
+    # add the title
+    plt.title("Proportion of STs across 11F_like samples in GPS")
+
+    # show the plot
+    plt.savefig("ST_plot.png", bbox_inches = 'tight')
+
+
+def plot_PCV_type_counts(df: pd.DataFrame):
+    # function to plot the percentage PCV type distribution across the dataset
+
+    # set the colour palette up with the number of colours equal to the number of unique PCVs
+    colours = sns.color_palette("bright", n_colors=len(pd.unique(df["PCV_type"])))
+
+    # create the plot
+    sns.countplot(
+        data=df,
+        x="PCV_type",
+        order=df["PCV_type"].value_counts().index,
+        palette=colours,
+        stat="percent",
+        hue="PCV_type",
+    )
+
+    # add the x and y axis labels
+    plt.xlabel("PCV_type")
+    plt.ylabel("Percentage")
+
+    # add the title
+    plt.title("Proportion of PCV_type across 11F_like samples in GPS")
+
+    # show the plot
+    plt.savefig("PCV_type_plot.png", bbox_inches = 'tight')
+
+
+def plot_Phenotypic_serotype_counts(df: pd.DataFrame):
+    # function to plot the percentage Phenotypic serotype distribution across the dataset
+
+    # set the colour palette up with the number of colours equal to the number of unique Phenotypic serotypes
+    colours = sns.color_palette("bright", n_colors=len(pd.unique(df["Phenotypic_serotype"])))
+
+    # create the plot
+    sns.countplot(
+        data=df,
+        x="Phenotypic_serotype",
+        palette=colours,
+        stat="percent",
+        hue="Phenotypic_serotype",
+    )
+
+    # add the x and y axis labels
+    plt.xlabel("Phenotypic_serotype")
+    plt.ylabel("Percentage")
+
+    # add the title
+    plt.title("Proportion of Phenotypic serotypes across 11F_like samples in GPS")
+
+    # show the plot
+    plt.savefig("Phenotypic_serotype_plot.png", bbox_inches = 'tight')
+
+
+def plot_Vaccine_period_counts(df: pd.DataFrame):
+    # function to plot the percentage Vaccine period distribution across the dataset
+
+    # set the colour palette up with the number of colours equal to the number of unique Vaccine periods
+    colours = sns.color_palette("bright", n_colors=len(pd.unique(df["Vaccine_period"])))
+
+    # create the plot
+    sns.countplot(
+        data=df,
+        x="Vaccine_period",
+        order=df["Vaccine_period"].value_counts().index,
+        palette=colours,
+        stat="percent",
+        hue="Vaccine_period",
+    )
+
+    # add the x and y axis labels
+    plt.xlabel("Vaccine_period")
+    plt.ylabel("Percentage")
+
+    # add the title
+    plt.title("Proportion of Vaccine periods across 11F_like samples in GPS")
+
+    # show the plot
+    plt.savefig("Vaccine_period_plot.png", bbox_inches = 'tight')
+
 # plot the GPSC counts
 plot_gpsc_counts(df)
+
```

Comment:
Can you put all of these function calls (`plot_gpsc_counts`, `plot_country_counts` etc..) into a function called `main`. We can then call the function `main()` to run all the plotting functions you've made at once.